gateway/handler_car.go

```diff
@@ -55,6 +96,7 @@ func (i *handler) serveCAR(ctx context.Context, w http.ResponseWriter, r *http.R
 	// Set Cache-Control (same logic as for a regular files)
 	addCacheControlHeaders(w, r, contentPath, rootCid)
 
```

Comment:
Perhaps related to what etag to use which we also need is what the name of the file in the attachment should be. Seems like it could be the root CID, or we could invent some syntax/transform to turn the request into a valid file name. Root cid certainly seems easier to implement although it makes it harder for the user to remember what was in the file (recall that the CAR header doesn't contain this information either).

Thoughts?

Reply:
what we're currently doing: `<root cid>.car` by default, also accepting a `filename` param to transform it https://github.com/filecoin-project/lassie/blob/main/pkg/server/http/ipfs.go#L149-L154

Reply:
SGTM, will use the root CID (at the top of the path, as opposed to before where it was the root CID of the terminal element).

Reply:
Hm.. I mean, the root cid makes sense as filename when there is no meaningful path segment. 
In other cases, we do have a meaningful label:
- `/ipfs/cid → cid.car`
- `/ipfs/cid/cats → cats.car` (makes way more sense than parent root cid)

Reply:
Minor quibble with this argument is that it's `root`->`cats`, not just `cats`; it'll retain that in the roots of the CAR but it's not _quite_ `cats` and the `root` in this case could be lower, or even higher in the DAG that leads to `cats`, so it is a bit significant. `<wikipedia cid>/wiki/Cat` as `Cat.car` might be nice, but there's multiple `<wikipedia cid>` values that lead there, so it's really "this particular wikipedia root -> Cat".

Could we combine? `<root>_<last-segment>.car` So `bafybeiaysi4s6lnjev27ln5icwm6tueaw2vdykrtjkwiphwekaywqhcjze/wiki/Cat` would yield `bafybeiaysi4s6lnjev27ln5icwm6tueaw2vdykrtjkwiphwekaywqhcjze_Cat.car`.

Reply:
Yes, it is sensible convention. 
Since we include all parent nodes, for deeper requests, let's include all parents:  `/ipfs/cid/wiki/Cat` → `cid_wiki_Cat.car`

Reply:
@lidel considering file names (and file paths) have a maximum length size in most modern OSes, I think it's sensible to only use the last element as @rvagg mentioned. At least then we ensure users won't have issues downloading files deeply nested. I think that's an unlikely event, but it may happen. Another option would be to cut the filename after 256 characters.

Reply:
Updated as `{CID}[_{last segment}].car`.